app/code/Magento/InventorySales/Plugin/CatalogInventory/StockManagement/ProcessRegisterProductsSalePlugin.php

```diff
@@ -147,4 +147,31 @@ private function checkItemsQuantity(array $items, array $productSkus, int $stock
             }
         }
     }
+
+    /**
+     * Build list of items connected to products with unsupported types.
+     *
+     * Remove items connected to products with unsupported product types and put them into separate array,
+     * in order to process them with standard behavior.
+     *
+     * @param array $items
+     * @param array $productSkus
+     * @return array
+     */
+    private function getExcludedItems(array &$items, array &$productSkus): array
```

Comment:
why do we accept parameters by reference? It's highly discouraged to do that

Reply:
@maghamed, I've removed references.